packages/rabbitmq/src/errorsSerialization.ts

```diff
@@ -0,0 +1,49 @@
+import { Cardano } from '@cardano-sdk/core';
+import { OutsideOfValidityInterval } from '@cardano-ogmios/schema';
+
+export interface ErrorDetails {
+  message?: string;
+}
+
+const TxSubmissionErrorsEntries = Object.entries(Cardano.TxSubmissionErrors) as [
+  keyof typeof Cardano.TxSubmissionErrors,
+  unknown
+][];
+
+export const deserializeErrors: Record<string, (details: ErrorDetails) => unknown> = {
+  unknown: (details) => new Error(details.message),
+  ...Object.fromEntries(
+    TxSubmissionErrorsEntries.map(([key]) => [
+      key,
+      (details: ErrorDetails) => new Cardano.TxSubmissionErrors[key](JSON.parse(details.message!))
+    ])
+  )
+};
+
+export interface ErrorDescription extends ErrorDetails {
```

Comment:
Could use [serialization utils](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/util/src/serializableObject.ts#L26) from `util` package.

Don't think you need to serialize `type` separately as the serialized error object contains it (`name`)

You could then deserialize into correct `instanceof` as it has [an option](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/util/src/serializableObject.ts#L19) for custom error prototype. Correct prototype is required for `isTxSubmissionError` to work and is lost with `JSON.stringify(err)`

Reply:
I learned only now the existence of this util; thank you.
I'll check it.

Reply:
You could use [this snippet](https://github.com/input-output-hk/cardano-js-sdk/blob/master/packages/cardano-services-client/src/TxSubmitProvider/txSubmitHttpProvider.ts#L12-L17) for `getErrorPrototype?: GetErrorPrototype;` in the serialization util, just instead of setting error prototype return it

Reply:
Could you please check this again, @mkazlauskas ?
Thank you

Reply:
Posted another review, let's continue this conversation in new comment threads